fetch-utils.js

```diff
@@ -26,15 +25,64 @@ export async function getDefaultTown() {
             {
                 name: 'Arcadia',
                 land: 1,
-                castle: 2,
+                castle: 1,
                 creature: 1,
                 message: []
             }
         ]);
+    
+    return checkError(response);
+}
+
+export async function updateName(newName) {
+    const user = await getUser();
+    
+    const response = await client
+        .from('builder')
+        .update({ name : newName })
+        .match({ user_id: user.user.id })
+        .single();
+    
+    return checkError(response);
+}
+//castle, land, creature, message update
+export async function updateLand(newId) {
+    const user = await getUser();
+
+    const response = await client
+        .from('builder')
+        .update({ land: newId })
+        .match({ user_id: user.user.id })
+        .single();
+    
+    return checkError(response);
+}
+
+export async function updateCastle(newId) {
+    const user = await getUser();
+
+    const response = await client
+        .from('builder')
+        .update({ castle: newId })
+        .match({ user_id: user.user.id })
+        .single();
 
     return checkError(response);
 }
 
+export async function updateCreature(newId) {
+    const user = await getUser();
+
+    const response = await client
+        .from('builder')
+        .update({ creature: newId })
+        .match({ user_id: user.user.id })
+        .single();
+
+    return checkError(response);
+}
+
```

Comment:
looks like you are missing your `updateSlogans(newSlogans)` function.  Write this so you can call it in your slogan event listener to update your slogan list.